File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    int n = 0;
    //prompt for dollars of non-zero positive value
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars <= 0);
    //round dollars to int cents
    int cents = round(dollars * 100);
    //determine number of coins
    while (cents >= 25)
    {
        cents -= 25;
        n++;
    }
    while (cents >= 10)
    {
        cents -= 10;
        n++;
    }
    while (cents >= 5)
    {
        cents -= 5;
        n++;
    }
    while (cents >= 1)
    {
        cents -= 1;
        n++;
    }
    //print number of coins
    printf("%i\n", n);
}

File: cash.py
from cs50 import get_float

# Define num of coin
n = 0

# Get +ve change from user and convert to cents
while True:
    change = int(get_float("Change owed: ") * 100)
    if change > 0:
        break


# Change of $0.25
while change >= 25:
    change -= 25
    n += 1

# Change of $0.10
while change >= 10:
    change -= 10
    n += 1

# Change of $0.05    
while change >= 5:
    change -= 5
    n += 1

# Change of $0.01   
while change >= 1:
    change -= 1
    n += 1

# Print total coins
print(f"{n}")